crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -614,6 +614,110 @@ impl<'w, 's> Commands<'w, 's> {
         self.queue(insert_or_spawn_batch(bundles_iter));
     }
 
+    /// Pushes a [`Command`] to the queue for adding a bundle to a batch of entities.
+    ///
+    /// `bundles_iter` is a type that can be converted into an ([`Entity`], [`Bundle`]) iterator
+    /// (it can also be a collection).
+    ///
+    /// When the command is applied, for each (`Entity`, `Bundle`) pair in the given `bundles_iter`,
+    /// the `Bundle` is added to the `Entity`, overwriting any existing components shared by the `Bundle`.
+    ///
+    /// This method is equivalent to iterating `bundles_iter`,
+    /// calling [`entity`](Self::entity) for each bundle,
+    /// and passing it to [`insert`](EntityCommands::insert),
+    /// but it is faster due to memory pre-allocation.
+    ///
+    /// # Panics
+    ///
+    /// This command panics if any of the given entities do not exist.
+    ///
+    /// See [`try_insert_batch`](Self::try_insert_batch) for the non-panicking version.
+    #[track_caller]
+    pub fn insert_batch<I, B>(&mut self, bundles_iter: I)
+    where
+        I: IntoIterator<Item = (Entity, B)> + Send + Sync + 'static,
+        B: Bundle,
+    {
+        self.queue(insert_batch(bundles_iter));
```

Comment:
Since the inner `insert_batch()` function is private, can it just be inlined? I'm not sure if it would break change detection tracking, but I would prefer this simpler form.

```suggestion
        #[cfg(feature = "track_change_detection")]
        let caller = Location::caller();
    
        self.queue(move |world: &mut World| {
            world.insert_batch_with_caller(
                bundles_iter,
                InsertMode::Replace,
                #[cfg(feature = "track_change_detection")]
                caller,
            );
        });
```

Reply:
That's how the other commands are set up, I just copied it lol. I guess it's something to make commands nicer on the user-side, but I dunno

Reply:
Ok, then keep it as it is!